notes-0414-universes2.tex

```diff
@@ -0,0 +1,130 @@
+
+\section{Glue Types}
+Since \texttt{Glue} types are types, they must be defined in the same way as all other types. The formation rule for a \texttt{Glue} type is:
+\begin{prooftree*}
+	\hypo{A : \U}
+	\hypo{\phi \vdash T : \U}
+	\hypo{\phi \vdash E : T \simeq A}
+	\infer3{\Glue \; [\phi \mapsto (T, E)] \; A : \U \; [\phi \mapsto T]}
+\end{prooftree*}
+This means that if we have a type $A : \U$ and this type is equivalent to the partial type $T$ under the constraint $\phi$, we can construct a \texttt{Glue} type that is judgementally equal to the type $T$ under the constraint $\phi$. \\
+
+The introduction rule defines the only constructor for the \texttt{Glue} type called '\texttt{glue}'. It is as follows:
+\begin{prooftree*}
+	\hypo{\phi \vdash t : T}
+	\hypo{a : A \; [\phi \mapsto \text{fst} \; E \; t]}
+	\infer2{\glue \; [\phi \mapsto t] \; a : (\Glue \; [\phi \mapsto (T, E)] \; A) \; [\phi \mapsto t]}
+\end{prooftree*}
+This rule takes two arguments: an element of the partial type $T$ and a total element of the type $A$ which agrees with the image of $t$ judgementally. \\
+
+The eliminator for a \texttt{Glue} type is called '\texttt{unglue}' and returns the base type that is equivalent to the partial element. More formally:
+\begin{prooftree*}
+	\hypo{u : \Glue \; [\phi \mapsto (T,E)] \; A}
+	\infer1{\unglue \; [\phi \mapsto E] \; u : A \; [\phi \mapsto \text{fst} \; E \; u]}
+\end{prooftree*}
+
+Applying the eliminator to the constructor and the constructor to the eliminator will work as expected. In the first case: $$\unglue \; [\phi \mapsto E] \; (\glue \; [\phi \mapsto t] \; a) \equiv a : A$$ and in the second case: $$u \equiv \glue \; [\phi \mapsto u] \; (\unglue \; [\phi \mapsto E] \; u) : \Glue \; [\phi \mapsto (T, E)] \; A$$
+\newpage
+
+\section{Composition of Glue Types}
+We will also need to define the Kan operators for the \texttt{Glue} types. For the homogeneous composition we can construct a 3-cube where the top face has the first \texttt{Glue} type on the front edge (M) and the second \texttt{Glue} type as the left wall (N). This means that the corresponding edges on the bottom face will have the total elements retrieved by applying the eliminator on the two \texttt{Glue} types. To get the result of the homogeneous composition, we need to call hfill on M (1). This will give a line that we can use for the right wall of the top face. By applying \texttt{unglue} to this result, we can get the entire right face (2), at which point we have enough information to perform homogeneous composition on the bottom face of the cube (3). This will give us a \texttt{Glue} type in the back face, since we have the two points on top as well as the bottom edge. 
```

Comment:
This is the one of the longer paragraphs. The picture can be made floating or put before this paragraph.